pkg/landscaper/installations/helper.go

```diff
@@ -449,3 +450,57 @@ func ResolveConfigMapReference(ctx context.Context, kubeClient client.Client, co
 
 	return completeData, data, cm.Generation, nil
 }
+
+// HandleSubComponentPhaseChanges updates the phase of the given installation, if its phase doesn't match the combined phase of its subinstallations/executions anymore
+func HandleSubComponentPhaseChanges(ctx context.Context, kubeClient client.Client, inst *lsv1alpha1.Installation, getOperation func(context.Context, *lsv1alpha1.Installation) (*Operation, error), getExports func(context.Context, *Operation) ([]*dataobjects.DataObject, []*dataobjects.Target, error), logger logr.Logger) error {
+	execRef := inst.Status.ExecutionReference
+	phases := []lsv1alpha1.ComponentInstallationPhase{}
+	if execRef != nil {
+		exec := &lsv1alpha1.Execution{}
+		err := kubeClient.Get(ctx, execRef.NamespacedName(), exec)
+		if err != nil {
+			return fmt.Errorf("error getting execution for installation %s/%s: %w", inst.Namespace, inst.Name, err)
+		}
+		phases = append(phases, lsv1alpha1.ComponentInstallationPhase(exec.Status.Phase))
+	}
+	subinsts, err := listSubinstallations(ctx, kubeClient, inst)
+	if err != nil {
+		return fmt.Errorf("error fetching subinstallations for installation %s/%s: %w", inst.Namespace, inst.Name, err)
+	}
+	for _, sub := range subinsts {
+		phases = append(phases, sub.Status.Phase)
+	}
+	cp := lsv1alpha1helper.CombinedInstallationPhase(phases...)
+	if inst.Status.Phase != cp {
+		// Phase is completed but doesn't fit to the deploy items' phases
+
+		// get operation
+		instOp, err := getOperation(ctx, inst)
+		if err != nil {
+			return fmt.Errorf("unable to construct operation for installation %s/%s: %w", inst.Namespace, inst.Name, err)
+		}
+		logger.V(5).Info("execution phase mismatch", "phase", string(inst.Status.Phase), "combinedPhase", string(cp))
+		err = instOp.UpdateInstallationStatus(ctx, inst, cp)
+		if err != nil {
+			return fmt.Errorf("error updating installation status for installation %s/%s: %w", inst.Namespace, inst.Name, err)
+		}
+
+		// recompute exports
+		dataExports, targetExports, err := getExports(ctx, instOp)
```

Comment:
this is only needed if we are in succeeded state.
With that code we would also update the exports during every other state e.g. `Progressing` or `Failed`